### PR TITLE
Fixes validation of input bytes in the Base64 decoder

### PR DESCRIPTION
**Pull request by fenik17**
Motivation:
In the current implementation of Base64 decoder an invalid character `½` (`\u00BD`) is treated as `=`. Also the character `\u007F` leads to `ArrayIndexOutOfBoundsException`.

Modification:
Explicitly checks that all input bytes are ASCII characters (greater than zero). Fix decodabet tables.

Result:
Correctly validation input bytes in Base64 decoder.

### CODE REVIEW

**Comment by fenik17** [comment]
@njhill thanx! I also fixed an `ArrayIndexOutOfBoundsException` and squashed commits.

**Comment by normanmaurer** [comment]
@netty-bot test this pleas

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: codec/src/test/java/io/netty/handler/codec/base64/Base64Test.java [hunk 1]
```diff
@@ -169,4 +169,24 @@ public void testOverflowEncodedBufferSize() {
     public void testOverflowDecodedBufferSize() {
         assertEquals(1610612736, Base64.decodedBufferSize(Integer.MAX_VALUE));
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void decodingFailsOnInvalidInputByte() {
+        Base64.decode(Unpooled.wrappedBuffer("eHh4\u00BD".getBytes(CharsetUtil.ISO_8859_1)));
```
Comment (normanmaurer) [inline codec/src/test/java/io/netty/handler/codec/base64/Base64Test.java]: Please ensure to release all the created ByteBuf instances in a finally block (true for all added tests)

File: codec/src/test/java/io/netty/handler/codec/base64/Base64Test.java [hunk 2]
```diff
@@ -169,4 +169,24 @@ public void testOverflowEncodedBufferSize() {
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void decodingFailsOnInvalidInputByte2() {
+        Base64.decode(Unpooled.wrappedBuffer("eHh4\u0080".getBytes(CharsetUtil.ISO_8859_1)));
```
Comment (normanmaurer) [inline codec/src/test/java/io/netty/handler/codec/base64/Base64Test.java]: You could also use copiedBuffer(...) to reduce some boilerplate

**Comment by fenik17** [comment]
@normanmaurer test fixed

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by fenik17** [comment]
@normanmaurer rebased

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Pull request merged**